### ISSUE TITLE
How to change plugin name in Ultrahand menu?

### ISSUE THREAD

**Issue by niheaven** [ppkantorski/Ultrahand-Overlay#168]
Hi,

I want to give meaningful names to my .ovl plugins in Ultrahand menu, how could I do this?

Thanks.

**Comment by lsilvatti** [comment]
I have the same question, plus, I don't know how to put translations, think that a tutorial on this might be a good idea, the project seems little documented

**Comment by ppkantorski** [comment]
Its not a feature, but you can hex it in (or modify the source code of the plugin).

However if it is a desired feature, then sure I can add in a way to override the name (if provided with user specified names) for the overlay files.

**Comment by lsilvatti** [comment]
> Its not a feature, but you can hex it in (or modify the source code of the plugin).  The name is currently being read from the compiled `.ovl` file.
> 
> To hex using an Ultrahand command, you can do something like this.
> 
> ```
> [Replace Overlay Text]
> hex-by-string <overlay_file_path> <text_to_replace_1> <text_replacement_1>
> hex-by-string <overlay_file_path> <text_to_replace_2> <text_replacement_2>
> hex-by-string <overlay_file_path> <text_to_replace_3> <text_replacement_3>
> hex-by-string <overlay_file_path> <text_to_replace_4> <text_replacement_4>
> hex-by-string <overlay_file_path> <text_to_replace_5> <text_replacement_5>
> ...
> ```
> 
> However if it is a desired feature, then sure I can add in a way to override the name (if provided with user specified names) for the overlay files.

Man, i think that a doc on overlays.ini and a property like "displayName" on that file that changes the name of the overlay will be neat, i really missed more documentation, like how to translate the overlays themselves too :)
Besides, I'm not criticizing, just suggesting. You are doing a tremendous and amazing work

**Comment by niheaven** [comment]
It's a desired feature, at least for me. It's useful for translation and I think a proper name should be fine for some plugins. And yes, I'm searching for plugin translation in Google but find nothing lol.

**Comment by ppkantorski** [comment]
> "how to translate the overlays themselves too"

Every overlay is compiled with text in whichever language.  It is up to overlay developers to translate the overlays (or the community).  Some overlays (like Ultrahand and a few others) have the ability to use translations that developers have implemented.  But Ultrahand itself has no control over this factor since every `.ovl` is built with its own bundle of libTesla.  When an overlay is opened, Ultrahand is closed before nx-ovlloader spawns the next overlay.  This means Ultrahand is no longer running when users are using an overlay. 

As for documentation on `overlays.ini` I'll add some in, but since all settings are managed by the system itself (besides adding in launch arguments) it wasn't essential to document.  Settings in `overlays.ini` can be set by clicking `Y` on an overlay on the menu.  `overlays.ini` is also located in `/config/ultrahand/`.

> It's a desired feature, at least for me. It's useful for translation and I think a proper name should be fine for some plugins. And yes, I'm searching for plugin translation in Google but find nothing lol.

Sure, ill add something like a new line `alternate_name=` for setting alternate names for overlays in the `overlays.ini` file.

**Comment by ppkantorski** [comment]
So on the dev fork for v1.6.9, ive added new options `custom_name=` and `custom_version=` so users can manually override the overlay names and versions with their own custom labels.  if left blank, it will display the stored name and version from the compiled overlay.

**Comment by niheaven** [comment]
Apologies for the additional question, but I encountered an issue while compiling the new version with divkitPro on macOS. The compilation process completed without errors; however, two problems arose: firstly, the output file is 1.6MB, and secondly, upon loading it onto the Switch, an error occurs just after the Nintendo logo disappears. What could be the cause?

**Comment by ppkantorski** [comment]
> Apologies for the additional question, but I encountered an issue while compiling the new version with divkitPro on macOS. The compilation process completed without errors; however, two problems arose: firstly, the output file is 1.6MB, and secondly, upon loading it onto the Switch, an error occurs just after the Nintendo logo disappears. What could be the cause?

It may be due to some of the newer lib updates getting larger.  To build, it is probably recommended for now to use specific versions for a couple of the `dkp-pacman` libs.  Give me a sec, I will double check which I am suing.

**Comment by ppkantorski** [comment]
```
devkitA64 r25.1-1 -> r26.1-2
switch-jansson 2.11-2 -> 2.11-3
switch-libarchive 3.7.2-5 -> 3.7.2-6
switch-libexpat 2.2.5-2 -> 2.2.5-3
switch-mbedtls 2.28.3-2 -> 2.28.3-3
switch-zlib 1.3-1 -> 1.3.1-1
switch-zziplib 0.13.69-3 -> 0.13.76-2
```

from what i remember, i believe devkitA64 was a really beefy update, and i saw no problems using the prior devkitA64.  i will have to go through these specific libraries again and make sure that it doesnt destroy the compilation size. i am touching the edge the allowable size for the overlay, any larger than the current size will start to cause problems.

**Comment by niheaven** [comment]
Thank you very much! So the error may be related to the building of libraries? I will sync with your settings and attempt to build it again.

**Comment by ppkantorski** [comment]
I optimized the code a bit more, was able to reduce size slightly.  i also noticed downloads suddenly stopped working (potentially broken SSL) in v1.6.8 (not sure if everyone is now having this problem suddenly) but ive resolved it on this compilation.

[compiled_1.6.9-pre.zip](https://github.com/user-attachments/files/16607809/compiled_1.6.9-pre.zip)

**Comment by ppkantorski** [comment]
ok i dug into downloads a bit more.  Ive found that SSL verification will fail if time is slightly out of sync (even by a little), so users may need to use QuickNTP to sync their time if this happens.  First time it happened to me, so just a heads up.  I'll probably get around to releasing 1.6.9 soon after ive looked over a couple other things.

**Comment by niheaven** [comment]
Hi, the attached ovl works fine for me, but the self-compiled one still failed... So it is devkitPro tools's incompatible now.

**Comment by ppkantorski** [comment]
Do let me know if you are still having problems, you could raise another issue.  I will be looking over the rest of the dkp-pacman updates sometime when I get the chance.  But seeing how I have resolved the current topic with the release of v1.6.9, I will be closing this particular issue.  Hopefully this update is helpful.
